spec/serialization_spec.rb

```diff
@@ -43,6 +45,505 @@ def self.name
       expect(json['data'][0]['attributes']).to eq('first_name' => 'John')
     end
 
+    describe 'types' do
+      # can coerce everything...
+      context 'when string' do
+        before do
+          resource.attribute :age, :string
+        end
+
+        it 'coerces' do
+          PORO::Employee.create(age: 1)
+          render
+          expect(attributes['age']).to eq('1')
+        end
+      end
+
+      context 'when integer' do
+        before do
+          resource.attribute :age, :integer
+        end
+
+        it 'coerces strings' do
+          PORO::Employee.create(age: '40')
+          render
+          expect(attributes['age']).to eq(40)
+        end
+
+        it 'allows nils' do
+          PORO::Employee.create(age: nil)
+          render
+          expect(attributes['age']).to eq(nil)
+        end
+
+        context 'when cannot coerce' do
+          before do
+            PORO::Employee.create(age: 'foo')
+          end
+
+          # Note ArgumentError not ConstraintError
+          # Unfortunate but not really our purview
+          it 'raises error' do
+            expect {
+              render
+            }.to raise_error(ArgumentError)
+          end
+        end
+      end
+
+      # NB: json is string because json can't support
+      # the level of precision BigDecimal requires
+      context 'when decimal' do
+        before do
+          resource.attribute :age, :decimal
+        end
+
+        it 'coerces integers' do
+          PORO::Employee.create(age: 40)
+          render
+          expect(attributes['age']).to eq('0.4E2')
+        end
+
+        it 'coerces strings' do
+          PORO::Employee.create(age: '40.01')
+          render
+          expect(attributes['age']).to eq('0.4001E2')
+        end
+
+        it 'allows nils' do
+          PORO::Employee.create(age: nil)
+          render
+          expect(attributes['age']).to eq(nil)
+        end
+
+        # Unfortunate we don't get a better error class
+        context 'when cannot coerce' do
+          before do
+            PORO::Employee.create(age: {})
+          end
+
+          it 'raises error' do
+            expect {
+              render
+            }.to raise_error(NoMethodError)
+          end
+        end
+      end
+
+      context 'when float' do
+        before do
+          resource.attribute :age, :float
+        end
+
+        it 'coerces strings' do
+          PORO::Employee.create(age: '40.01')
+          render
+          expect(attributes['age']).to eq(40.01)
```

Comment:
From a usability perspective, I'm wondering whether it makes sense to collapse `float` and `decimal` types into a single thing, represented as a string?  I imagine myself getting them mixed up and misusing them more than being specific about the one I need.  If we don't want to collapse, I'd be more in favor of calling out `decimal` a bit more explicitly and using `big_decimal` or whatever instead.

Reply:
It's different behavior. A BigDecimal will always render as a string, a float will often render correctly. 90% of the time you'd want Float.

Reply:
I'd be fine with :big_decimal though

Reply:
Yeah I don't really care which way we go. Prefer if we can unify, but if not for reasons like you state, then calling it out as special is better.  How would you feel about a less ruby-specific name like `float_as_string`?